docs/nnx/bridge_guide.md

```diff
@@ -0,0 +1,544 @@
+  def __init__(self, din, dout, rngs):
+    self.w = nnx.Param(nnx.initializers.lecun_normal()(rngs.params(), (din, dout)))
+    self.lora = nnx.LoRA(din, 3, dout, rngs=rngs)
+    self.count = Count(jnp.array(0))
+
+  def __call__(self, x):
+    self.count += 1
+    return (x @ self.w.value) + self.lora(x)
+
+xkey, pkey, dkey = jax.random.split(jax.random.key(0), 3)
+x = jax.random.normal(xkey, (2, 4))
+model = bridge.to_linen(NNXMultiCollections, 4, 3)
+var = model.init({'params': pkey, 'dropout': dkey}, x)
+print('All Linen collections:', list(var.keys()))
+print(var['params'])
+```
+
+    All Linen collections: ['nnx', 'LoRAParam', 'Count', 'params']
+    {'w': NNXMeta(var_type=<class 'flax.nnx.nnx.variables.Param'>, value=Array([[ 0.2916921 ,  0.22780475,  0.06553137],
+           [ 0.17487915, -0.34043145,  0.24764155],
+           [ 0.6420431 ,  0.6220095 , -0.44769976],
+           [ 0.11161668,  0.83873135, -0.7446058 ]], dtype=float32), metadata={'get_value_hooks': (), 'set_value_hooks': (), 'create_value_hooks': (), 'add_axis_hooks': (), 'remove_axis_hooks': ()})}
+
+
+## Partition metadata
+
+Flax uses a metadata wrapper box over the raw JAX array to annotate how a variable should be sharded. 
+
+In Linen, this is an optional feature that triggered by using `nn.with_partitioning` on initializers (see more on [Linen partition metadata guide](https://flax.readthedocs.io/en/latest/guides/parallel_training/flax_on_pjit.html)). In NNX, since all NNX variables are wrapped by `nnx.Variable` class anyway, that class will hold the sharding annotations too. 
+
+The `bridge.ToNNX` and `bridge.ToLinen` API will automatically convert the sharding annotations, if you use the built-in annotation methods (aka. `nn.with_partitioning` for Linen and `nnx.with_partitioning` for NNX).
+
+### Linen to NNX
+
+Even if you are not using any partition metadata in your Linen module, the variable JAX arrays will be converted to `nnx.Variable`s that wraps the true JAX array within. 
+
+If you use `nn.with_partitioning` to annotate your Linen module's variables, the annotation will be converted to a `.sharding` field in the corresponding `nnx.Variable`. 
+
+You can then use `nnx.with_sharding_constraint` to explicitly put the arrays into the annotated partitions within a `jax.jit`-compiled function, to initialize the whole model with every array at the right sharding.
+
+
+```python
+class LinenDotWithPartitioning(nn.Module):
+  out_dim: int
+  @nn.compact
+  def __call__(self, x):
+    w = self.param('w', nn.with_partitioning(nn.initializers.lecun_normal(), ('in', 'out')), 
+                   (x.shape[-1], self.out_dim))
+    return x @ w
+
+@jax.jit
+def create_sharded_nnx_module(x):
+  model = bridge.lazy_init(bridge.ToNNX(LinenDotWithPartitioning(64), rngs=nnx.Rngs(0)), x)
+  static, state = nnx.split(model)
+  sharded_state = nnx.with_sharding_constraint(state, nnx.get_partition_spec(state))
+  return static, sharded_state
```

Comment:
Consider using `nnx.jit` here so you don't have to use `merge` later. 
```suggestion
@nnx.jit
def create_sharded_nnx_module(x):
  model = bridge.lazy_init(bridge.ToNNX(LinenDotWithPartitioning(64), rngs=nnx.Rngs(0)), x)
  state = nnx.state(model)
  sharded_state = nnx.with_sharding_constraint(state, nnx.get_partition_spec(state))
  nnx.update(model, sharded_state)
  return model
```

Reply:
Done. I should get more used to it!